eyeball-im-util/src/vector/ops.rs

```diff
@@ -14,30 +20,30 @@ pub trait VectorDiffContainerOps<T>: Sized {
         f: impl FnMut(VectorDiff<T>) -> Option<VectorDiff<U>>,
     ) -> Option<VectorDiffContainerFamilyMember<Self::Family, U>>;
 
-    fn push_into_limit_buf(
+    fn push_into_array_buf(
         self,
-        buffer: &mut Self::LimitBuf,
-        make_diffs: impl FnMut(VectorDiff<T>) -> ArrayVec<VectorDiff<T>, 2>,
+        buffer: &mut Self::ArrayBuf,
+        make_diffs: impl FnMut(VectorDiff<T>) -> ArrayVec<VectorDiff<T>, BUF_CAP>,
     ) -> Option<Self>;
 
-    fn pop_from_limit_buf(buffer: &mut Self::LimitBuf) -> Option<Self>;
+    fn pop_from_array_buf(buffer: &mut Self::ArrayBuf) -> Option<Self>;
 
-    fn push_into_sort_buf(
+    fn push_into_vec_buf(
         self,
-        buffer: &mut Self::SortBuf,
-        make_diffs: impl FnMut(VectorDiff<T>) -> SmallVec<[VectorDiff<T>; 2]>,
+        buffer: &mut Self::VecBuf,
+        make_diffs: impl FnMut(VectorDiff<T>) -> SmallVec<[VectorDiff<T>; BUF_CAP]>,
     ) -> Option<Self>;
 
-    fn pop_from_sort_buf(buffer: &mut Self::SortBuf) -> Option<Self>;
+    fn pop_from_vec_buf(buffer: &mut Self::VecBuf) -> Option<Self>;
 }
 
 #[allow(unreachable_pub)]
 pub type VectorDiffContainerFamilyMember<F, U> = <F as VectorDiffContainerFamily>::Member<U>;
 
 impl<T> VectorDiffContainerOps<T> for VectorDiff<T> {
     type Family = VectorDiffFamily;
-    type LimitBuf = Option<VectorDiff<T>>;
-    type SortBuf = SmallVec<[VectorDiff<T>; 2]>;
+    type ArrayBuf = Option<VectorDiff<T>>;
+    type VecBuf = SmallVec<[VectorDiff<T>; BUF_CAP]>;
```

Comment:
Oh, now I remember why I didn't like this initially. These names only make sense for `impl<T> VectorDiffContainerOps<T> for Vec<VectorDiff<T>>`, not for `impl<T> VectorDiffContainerOps<T> for VectorDiff<T>`... Honestly, I'm back to _strongly_ disliking the new names 😕